kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterConstructContext.java

```diff
@@ -6,21 +6,20 @@
 
 package io.kroxylicious.proxy.filter;
 
-import io.kroxylicious.proxy.config.BaseConfig;
 import io.kroxylicious.proxy.service.Context;
 
-public interface FilterConstructContext extends Context {
+public interface FilterConstructContext<B> extends Context<B> {
     FilterExecutors executors();
 
-    static FilterConstructContext wrap(BaseConfig config, FilterExecutors executors) {
-        return new FilterConstructContext() {
+    static <B> FilterConstructContext<B> wrap(B config, FilterExecutors executors) {
```

Comment:
These `wrap` methods are implicitly `public`, which seems to be leaking an implementation detail. Filter authors shouldn't know or care how the context provided to them is constructed.